frontend/src/app/core/menu/user-menu/user-menu.component.ts

```diff
@@ -11,26 +10,29 @@ import _ from 'underscore';
   styleUrls: ['./user-menu.component.scss']
 })
 export class UserMenuComponent {
-    isLoading$: Observable<boolean>;
-    user$: Observable<User | null | undefined>;
-    showSignIn$: Observable<boolean>;
+    public isLoading$ = this.authService.currentUser$.pipe(
+        map(_.isUndefined)
+    );
 
-    icons = {
+    public user$ = this.authService.currentUser$;
+
+    public showSignIn$ = this.authService.currentUser$.pipe(
+        map(_.isNull)
+    );
+
+    public icons = {
         user: faUser,
     };
 
-    constructor(private authService: AuthService) {
-        this.isLoading$ = this.authService.currentUser$.pipe(
-            map(_.isUndefined)
-        );
-        this.user$ = this.authService.currentUser$;
-        this.showSignIn$ = this.authService.currentUser$.pipe(
-            map(_.isNull)
-        );
-    }
+    public loading$ = merge(
+        this.authService.logout$.pipe(map(() => true)),
+        this.authService.logoutResult$.pipe(map(() => false)),
+    ).pipe(startWith(false));
```

Comment:
It's confusing that this component has both `isLoading$` and `loading$`. Please rename to something like `authLoading$` and `logoutLoading$`.

Reply:
Done!